sdk/attestation/azure-security-attestation/tests/test_policy_getset_async.py

```diff
@@ -17,273 +17,387 @@
-        await self._test_get_policy_management_certificates(attestation_isolated_url, attestation_isolated_signing_certificate)
+            assert len(policy_signers) == 0
 
-    @AttestationPreparer()
-    async def test_aad_get_policy_management_certificates(self, attestation_aad_url):
-        await self._test_get_policy_management_certificates(attestation_aad_url, None)
+    @staticmethod
+    def is_isolated_url(instance_url, **kwargs):
+        # type: (str, Any) -> bool
+        return instance_url == kwargs.get("attestation_isolated_url")
 
+    @pytest.mark.live_test_only
     @AttestationPreparer()
-    async def test_shared_get_policy_management_certificates(self, attestation_location_short_name):
-        await self._test_get_policy_management_certificates(self.shared_base_uri(attestation_location_short_name), None)
+    @AllInstanceTypes
+    async def test_get_policy_management_certificates(self, **kwargs):
+        instance_url = kwargs.pop("instance_url")
+        expected_certificate = None
+        if self.is_isolated_url(instance_url, **kwargs):
+            expected_certificate = pem_from_base64(
+                kwargs.get("attestation_isolated_signing_certificate"), "CERTIFICATE"
+            )
+        await self._test_get_policy_management_certificates(
+            instance_url, expected_certificate
+        )
 
-    @AttestationPreparer()
     @pytest.mark.live_test_only
+    @AttestationPreparer()
     async def test_add_remove_policy_certificate(
-        self, 
+        self,
         attestation_isolated_url,
         attestation_isolated_signing_certificate,
         attestation_isolated_signing_key,
         attestation_policy_signing_key0,
-        attestation_policy_signing_certificate0):
-        #type:(str, str, str, str, str, str) -> None
-
-        der_signing_cert = base64.b64decode(attestation_isolated_signing_certificate)
-        der_signing_key = base64.b64decode(attestation_isolated_signing_key)
-        signing_key = AttestationSigningKey(der_signing_key, der_signing_cert)
-
-        admin_client = self.create_admin_client(attestation_isolated_url)
+        attestation_policy_signing_certificate0,
+    ):
+        # type: (str, str, str, str, str, str) -> None
+
+        signing_certificate = pem_from_base64(
+            attestation_isolated_signing_certificate, "CERTIFICATE"
+        )
+        key = pem_from_base64(attestation_isolated_signing_key, "PRIVATE KEY")
+
+        pem_certificate_to_add = pem_from_base64(
+            attestation_policy_signing_certificate0, "CERTIFICATE"
+        )
+
+        admin_client = self.create_admin_client(
+            attestation_isolated_url,
+            signing_key=key,
+            signing_certificate=signing_certificate,
+        )
+
+        # Test 0 positional parameters, should throw.
+        with pytest.raises(TypeError):
```

Comment:
Just something to consider: if there's important information you want to be in the exception message, you can do this and make assertions about the content of `ex`
```suggestion
        with pytest.raises(TypeError) as ex:
```